hermeto/core/models/input.py

```diff
@@ -120,6 +122,60 @@ def _validate_ssl_options(self) -> Self:
         return self
 
 
+class BinaryFilter(pydantic.BaseModel, extra="forbid"):
+    """Represents a filter for binary packages.
+
+    An empty filters set means "accept all" (no filtering).
+    A populated filters set means "accept only these specific items".
+    """
+
+    filters: set[str] = set()
+
+    @property
+    def is_all(self) -> bool:
+        """Return True if this filter accepts all (no filtering)."""
+        return not self.filters
+
+    def __str__(self) -> str:
+        return ":all:" if self.is_all else ",".join(sorted(self.filters))
+
+    def __repr__(self) -> str:
+        return (
+            f"<{type(self).__name__} "
+            f"filters={':all:' if self.is_all else ','.join(sorted(self.filters))}>"
+        )
+
+
+def _parse_binary_filter(value: Any) -> BinaryFilter:
+    """Parse either None, BinaryFilter, or comma-separated string into a BinaryFilter object."""
+    if value is None:
+        return BinaryFilter()
+    if isinstance(value, BinaryFilter):
+        return value
+    if not isinstance(value, str):
+        raise ValueError(
+            (
+                f"Binary filter must be a string (e.g., 'x86_64,aarch64' or ':all:'). "
+                f"Got type: {type(value).__name__}"
+            )
+        )
+
+    value = value.strip()
+    if value == BINARY_FILTER_ALL:
+        return BinaryFilter()
+
+    filters = {f.strip() for f in value.split(",") if f.strip()}
```

Comment:
nitpick: can we use walrus here? `{f for f in value.split(",") if (f := f.strip())}` (or with any other variable other than `f` :) ).

Reply:
Done